src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs

```diff
@@ -54,7 +50,14 @@ public virtual CSharpCompilationOptions CSharpCompilationOptions
             }
         }
 
-        public EmitOptions EmitOptions { get; }
+        public virtual EmitOptions EmitOptions
```

Comment:
Is this just virtual so people can override and customize if they want to?

Reply:
Yup. It prevented us from suggesting overriding the property as a workaround to specifying debug type.